sandbox/extract-imagery-metadata_stevendepaschalis.R

```diff
@@ -49,56 +28,73 @@ extract_metadata = function(exif) {
   # Remove NA and infinity values
   time_diff_seconds <- time_diff_seconds[is.finite(time_diff_seconds)]
 
-  # Calculate image frequency (images per second)
-  image_frequency <- 1 / time_diff_seconds
+  # Calculate median and median absolute deviation (MAD)
+  median_time_diff <- median(time_diff_seconds)
+  mad_time_diff <- mad(time_diff_seconds)
+
+  # Set threshold as median + 3 * MAD
+  threshold <- median_time_diff + 3 * mad_time_diff
+
+  # Exclude outliers
+  filtered_time_diff <- time_diff_seconds[time_diff_seconds <= threshold]
 
-  # Check if there are any non-finite image frequencies
-  if (any(!is.finite(image_frequency))) {
-    stop("Invalid image frequencies detected.")
-  }
+  # Calculate image frequency (images per second) from filtered values
+  image_frequency <- 1 / filtered_time_diff
 
-  # Calculate mean image frequency
+  # Calculate mean image frequency from filtered values
   mean_image_frequency <- mean(image_frequency, na.rm = TRUE)
 
+  return(mean_image_frequency)
+}
+
+# Define the function to extract resolution and aspect ratio
+extract_resolution_and_aspect_ratio <- function(exif) {
   # Get Xresolution and Yresolution from the EXIF data
-  resolution_x <- unique(exif$Xresolution)
-  resolution_y <- unique(exif$Yresolution)
+  resolution_x <- unique(exif$XResolution)
+  resolution_y <- unique(exif$YResolution)
+
+  # Calculate mode resolution
+  mode_resolution_x <- as.numeric(names(sort(table(resolution_x), decreasing = TRUE)[1]))
+  mode_resolution_y <- as.numeric(names(sort(table(resolution_y), decreasing = TRUE)[1]))
 
   # Get aspect ratio
-  aspect_ratio <- unique(exif$imagewidth / exif$imageheight)
+  aspect_ratio <- mode_resolution_x / mode_resolution_y
 
+  return(list(resolution_x = mode_resolution_x,
+              resolution_y = mode_resolution_y,
+              aspect_ratio = aspect_ratio))
+}
+
+# Define the function to extract image file type
+extract_file_type <- function(exif) {
   # Get image file format
   image_file_format <- unique(exif$FileType)
 
-  # Check if resolution attributes are identical across all images
-  if (length(resolution_x) > 1 | length(resolution_y) > 1) {
-    # If not identical, compute average
-    avg_resolution_x <- mean(resolution_x)
-    avg_resolution_y <- mean(resolution_y)
-
-    # Record a warning
-    warning("Resolution attributes are not identical across all images. Average resolution computed.")
-  } else {
-    # If identical, use the single values
-    avg_resolution_x <- resolution_x
-    avg_resolution_y <- resolution_y
-  }
-
-  # Check if image file format is consistent
-  if (length(image_file_format) > 1) {
-    # If not identical, return a warning
-    warning("Image file format varies across images.")
-  }
```

Comment:
I really liked this check you originally included. It would be great to add it back. Including this is not mutually exclusive with returning the mode image type.